### PR TITLE
feat(vaultFactory): UI version hash governed param

### LINKED ISSUE #6860: EC charter power to update value of well-known name for UI hosting
## What is the Problem Being Solved?

For decentralized dapp hosting (#4350), an elected body such as the EC needs the ability to update a hash under a well-known name such as `published.vaults.governance`.

## Description of the Design

Add a `uiVersionHash` to the vault factory governed parameters.

## Security Considerations

This is a solution to a bunch of security constraints explored in #4350 .

## Scaling Considerations

The on-chain work is like any other EC decision (IOU # of known issue there).

There's a bunch of off-chain fan-out, but that's outside the scope of this issue.

## Test Plan

unit test

### PR DESCRIPTION
**Pull request by dckc**
Add a governed param that allows the electorate to choose a version hash that a web client loader can use as a key into content-addressable storage such as IPFS.

fixes: #6860

### Security Considerations

A hash (example: `QmWGhGssVvEnG8b8TEftMSfE6HXsy96CXTNfB1QLWmYRcW`) is a bit opaque and subject to error, but techniques such as [jazzicons](https://www.npmjs.com/package/@metamask/jazzicon) can help with that.

### Scaling Considerations

consumes a small amount each time the governance values change

### Documentation Considerations

The EC should be trained to use it.

### Testing Considerations

 - Minimal unit test coverage.
 - [manual integration test using `agd query vstorage`](#issuecomment-1408988868)

### CODE REVIEW

**Review by Chris-Hibbert — APPROVED** [review]

File: packages/inter-protocol/src/vaultFactory/params.js [hunk 1]
```diff
@@ -23,20 +23,23 @@ export const LIQUIDATION_INSTALL_KEY = 'LiquidationInstall';
...
  * @param {Amount} shortfallInvitationAmount
+ * @param {string} [uiVersionHash]
  */
 const makeVaultDirectorParams = (
   electorateInvitationAmount,
   liquidationInstall,
   liquidationTerms,
   minInitialDebt,
   shortfallInvitationAmount,
+  uiVersionHash = 'UNDECIDED',
```
Comment (turadg) [inline packages/inter-protocol/src/vaultFactory/params.js]: let's not make it optional at all levels. the job of this function is just to transform the passed in values. it's sort of a legacy that it exists.

File: packages/inter-protocol/src/vaultFactory/params.js [hunk 2]
```diff
@@ -23,20 +23,23 @@ export const LIQUIDATION_INSTALL_KEY = 'LiquidationInstall';
 export const LIQUIDATION_TERMS_KEY = 'LiquidationTerms';
 export const MIN_INITIAL_DEBT_KEY = 'MinInitialDebt';
 export const SHORTFALL_INVITATION_KEY = 'ShortfallInvitation';
+export const ENDORSED_UI_KEY = 'EndorsedUI';
```
Comment (turadg) [inline packages/inter-protocol/src/vaultFactory/params.js:26]: note that this is `UI` not `Ui`. no strong opinion.

File: packages/inter-protocol/test/vaultFactory/test-vaultFactory.js [hunk 3]
```diff
@@ -2901,6 +2901,7 @@ test('governance publisher', async t => {
   t.is(current.LiquidationTerms.type, 'unknown');
   t.is(current.MinInitialDebt.type, 'amount');
   t.is(current.ShortfallInvitation.type, 'invitation');
+  t.is(current.UiVersionHash.type, 'string');
```
Comment (turadg) [inline packages/inter-protocol/test/vaultFactory/test-vaultFactory.js]: oops I missed this one @dckc .  approving since it'll have to be fixed to pass.

**Review by turadg — APPROVED** [review]
Recap from call:
- param name needed constrain how the specifier is encoded
- param name should say what the authority of the parameter is
  - unlike others it's not observed by any contracts
  - it's just an endorsement from the governing body for what the user should use

**Comment by dckc** [comment]
I did a manual integration test with `scenario2-run-chain-economy`:

```
$ agoric follow --output json :published.vaultFactory.governance | jq .current.EndorsedUI
{
  "type": "string",
  "value": "NO ENDORSEMENT"
}
```

**Pull request merged**